DUI3-DX/Converters/Revit/Speckle.Converters.RevitShared/ToSpeckle/WallConversionToSpeckle.cs

```diff
@@ -76,64 +89,73 @@ public override SOBR.RevitWall RawConvert(DB.Wall target)
-    if (hostedObjects.Count > 0)
+    List<Base> wallChildren = new();
+    if (target.CurtainGrid is DB.CurtainGrid grid)
+    {
+      wallChildren.AddRange(ConvertElements(grid.GetMullionIds()));
+      wallChildren.AddRange(ConvertElements(grid.GetPanelIds()));
+    }
+    else if (target.IsStackedWall)
     {
-      if (speckleWall.GetDetachedProp("elements") is List<Base> elements)
-      {
-        elements.AddRange(hostedObjects);
-      }
-      else
-      {
-        speckleWall.SetDetachedProp("elements", hostedObjects);
-      }
+      wallChildren.AddRange(ConvertElements(target.GetStackedWallMemberIds()));
     }
+    wallChildren.AddRange(ConvertElements(target.GetHostedElementIds()));
+    return wallChildren;
   }
 
-  private void AssignVoids(DB.Wall target, SOBR.RevitWall speckleWall)
+  private IEnumerable<Base> ConvertElements(IEnumerable<DB.ElementId> elementIds)
   {
-    List<DB.CurveArray> voids = GetWallVoids(target);
-    List<SOG.Polycurve> polycurves = voids.Select(v => _curveArrayConverter.RawConvert(v)).ToList();
+    foreach (DB.ElementId elementId in elementIds)
+    {
+      yield return _converter.Convert(_contextStack.Current.Document.Document.GetElement(elementId));
+    }
+  }
 
-    if (polycurves.Count > 0)
+  private void AssignDisplayValue(DB.Wall target, RevitWall speckleWall)
+  {
+    if (target.CurtainGrid is null)
     {
-      speckleWall["voids"] = polycurves;
+      speckleWall.displayValue = _displayValueExtractor.GetDisplayValue(target);
     }
+    speckleWall.displayValue = new List<SOG.Mesh>();
   }
 
-  private List<DB.CurveArray> GetWallVoids(DB.Wall wall)
+  private void AssignHostedElements(SOBR.RevitWall speckleWall, List<Base> hostedObjects)
   {
-    List<DB.CurveArray> curveArrays = new();
-    var profile = ((DB.Sketch)_contextStack.Current.Document.Document.GetElement(wall.SketchId))?.Profile;
+    if (hostedObjects.Count == 0)
+    {
+      return;
+    }
 
-    if (profile == null)
+    if (speckleWall.GetDetachedProp("elements") is List<Base> elements)
     {
-      return curveArrays;
+      elements.AddRange(hostedObjects);
     }
+    else
+    {
+      speckleWall.SetDetachedProp("elements", hostedObjects);
+    }
+  }
 
-    for (var i = 1; i < profile.Size; i++)
+  private void AssignVoids(DB.Wall target, SOBR.RevitWall speckleWall)
+  {
+    DB.CurveArrArray? profile = ((DB.Sketch)target.Document.GetElement(target.SketchId))?.Profile;
+    if (profile is null)
     {
-      var segments = profile.get_Item(i);
-      if (segments.Cast<DB.Curve>().Count() > 2)
-      {
-        curveArrays.Add(segments);
-      }
+      return;
     }
 
-    return curveArrays;
+    List<SOG.Polycurve> polycurves = _curveArrArrayConverter.RawConvert(profile);
+
+    if (polycurves.Count > 1)
+    {
+      // POC: we have been assuming that the first curve is the element and the rest of the curves are openings
+      // this isn't always true
+      speckleWall["voids"] = polycurves.Skip(1);
```

Comment:
We know for a fact many times these are not even `voids`... 😅 we really should re-evaluate the whole wall conversions in revit to account for sketched walls properly

Reply:
I agree. We also need to stop using dynamic properties (in my opinion)

Reply:
YES!! 🙌🏼 you've got my vote and @JR-Morgan's hahah

Reply:
Can w add a ticket to review "re-evaluate the whole wall conversions in revit to account for sketched walls properly" and drop the CNX here plz?